src/cmd_line/commands/read.ts

```diff
@@ -0,0 +1,81 @@
+"use strict";
+
+import * as node from "../node";
+import {readFile} from 'fs';
+import {exec} from 'child_process';
+import {TextEditor} from '../../textEditor';
+
+export interface IReadCommandArguments extends node.ICommandArgs {
+  file?: string;
+  cmd?: string;
+}
+
+//
+//  Implements :read and :read!
+//  http://vimdoc.sourceforge.net/htmldoc/insert.html#:read
+//  http://vimdoc.sourceforge.net/htmldoc/insert.html#:read!
+//
+export class ReadCommand extends node.CommandBase {
+  protected _arguments : IReadCommandArguments;
+
+  constructor(args : IReadCommandArguments) {
+    super();
+    this._name = 'read';
+    this._shortName = 'r';
+    this._arguments = args;
+  }
+
+  get arguments() : IReadCommandArguments {
+    return this._arguments;
+  }
+
+  async execute() : Promise<void> {
+    const textToInsert = await this.getTextToInsert();
+    if (textToInsert) {
+      await TextEditor.insert(textToInsert);
+    }
+  }
+
+  async getTextToInsert() : Promise<string> {
+    if (this.arguments.file && this.arguments.file.length > 0) {
+      return await this.getTextToInsertFromFile();
+    } else if (this.arguments.cmd && this.arguments.cmd.length > 0) {
+      return await this.getTextToInsertFromCmd();
+    } else {
+      throw Error('Invalid arguments');
+    }
+  }
+
+  async getTextToInsertFromFile() : Promise<string> {
+    // TODO: Substitute utf8 with current file encoding, couldn't find this anywhere in vscode's api.
```

Comment:
Is there an issue tracking this with VSCode? Some initial Google says there's some config at the workspace level where you can set the encoding. Maybe we can read that config?